src/changelog/linker/mapper/leadingv.go

```diff
@@ -0,0 +1,82 @@
+package mapper
+
+import (
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/Masterminds/semver"
+	"github.com/newrelic/release-toolkit/src/changelog"
+	"github.com/newrelic/release-toolkit/src/changelog/linker"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	checkTimeoutSeconds = 1
+)
+
+// LeadingVCheck wraps another mapper and performs a check over the result of Map. If the check is not
+// successful, it tries to generate it prepending a leading v to the version.
+type LeadingVCheck struct {
+	mapper    linker.Mapper
+	checkLink func(link string) bool
+}
+
+// WithLeadingVCheck returns a LeadingVCheck with the provided underlying mapper and a check function which
+// performs a request to the url corresponding to the link and check its status code.
+func WithLeadingVCheck(mapper linker.Mapper) *LeadingVCheck {
+	return &LeadingVCheck{mapper: mapper, checkLink: checkLinkResponse}
+}
+
+func (l *LeadingVCheck) Map(dep changelog.Dependency) string {
+	link := l.mapper.Map(dep)
+	if link == "" || l.checkLink(link) {
+		return link
+	}
+
+	newDep, changed := l.switchDepLeadingV(dep)
+	if !changed {
+		return ""
+	}
+
+	link = l.mapper.Map(newDep)
+	if l.checkLink(link) {
+		return link
+	}
+
+	return ""
+}
+
+func (l *LeadingVCheck) switchDepLeadingV(dep changelog.Dependency) (changelog.Dependency, bool) {
+	literal := dep.To.Original()
+	switchedLiteral := ""
+	if strings.HasPrefix(literal, "v") {
+		switchedLiteral = literal[1:]
```

Comment:
Slicing strings make me shiver a bit 😅 
```suggestion
		switchedLiteral = strings.TrimPrefix(literal, "v")
```

Reply:
Don't be afraid of slices! 😉 (Although, using `TrimPrefix` is way more readable)